.circleci/docker/build.sh

```diff
@@ -295,7 +295,11 @@ if [ -n "$CLANG_VERSION" ]; then
   if !(drun clang --version 2>&1 | grep -qF "clang version $CLANG_VERSION"); then
     echo "CLANG_VERSION=$CLANG_VERSION, but:"
     drun clang --version
-    exit 1
+    if [[ "$image" == *-rocm* ]]; then
+      echo "For ROCm builds, version mismatch expected."
```

Comment:
yuck

Reply:
Should this check just do something else in the case of rocm?

Like I'd expect it to check against some version or don't do the check at all.

Also, if this check is invalid, should we even do a `CLANG_VERSION` at all?

Reply:
If you're not going to use clang, just undefine `CLANG_VERSION` in the config

Reply:
Should we leave the ECR entry for  `pytorch-linux-xenial-rocm-py3.6-clang7` or create a new one without the clang7, `pytorch-linux-xenial-rocm-py3.6`?

Reply:
Discussed offline.  New ECR entry created for later.  For now, keep everything as-is (e.g., installing clang7 even though it isn't used) to expedite the process of getting ROCm CI switched to the new image.

A follow up PR will remove the clang7 install and rocm-specific work-around in the install script.